### ISSUE TITLE
EC2 Image Builder: Create a new version of launch template after successful build

### ISSUE THREAD

**Issue by Arrayly** [hashicorp/terraform-provider-aws#20739]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add the parameters **launch_template_id (string)** and **update_launch_template_version (bool)** in the **ami_distribution_configuration** block to instruct EC2 Image Builder to create a new version of the launch template that includes the latest AMI ID after the build completes. This is useful for updating an existing autoscaling group to use the new AMI.

Currently, when the resource is deployed, I have to manually go into the dashboard to set the launch template id in the distribution configuration.

### New or Affected Resource(s)

[aws_imagebuilder_distribution_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/imagebuilder_distribution_configuration)

### Potential Terraform Configuration

```hcl
resource "aws_imagebuilder_distribution_configuration" "example" {
  name = "example"

  distribution {
    ami_distribution_configuration {
         
           launch_template_id = "lt-xxxxxxxxxxxxxx"
           update_launch_template_version = true

    }

    region = "us-east-1"
  }
}
```

Thank you! :)

**Comment by jleonardolemos** [comment]
I am very interested in this feature. I am willing to try a PR but i don know how to start, just point me to the right direction and i will try.

**Comment by Arrayly** [comment]
@jleonardolemos
You may use [this](https://github.com/aws-samples/ec2-auto-scaling-instance-refresh-sample) to help you get started. I had to convert it to terraform for my own project and then ran into this issue. If you look closely in the **template.yaml** it has the following configuration which is missing in terraform:

```
LaunchTemplateConfigurations:
   - LaunchTemplateId: !Ref SampleLaunchTemplate
     AccountId: !Sub ${AWS::AccountId}
     SetDefaultVersion: true
```
I hope this gives you a better understanding.

**Comment by simoferr98** [comment]
are there any updates for this future integration?

**Comment by jleonardolemos** [comment]
@simoferr98 not for my side

**Comment by rmccarthy-ellevation** [comment]
Any updates on this?

**Comment by kamilturek** [comment]
Should have been closed via #22842.

**Comment by kamilturek** [comment]
Hey @ewbankkit, could we please close this issue? It has many upvotes but the argument has been already added to the resource.
